src/main/java/net/dv8tion/jda/internal/entities/ActionMetadataImpl.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.entities;
+
+import net.dv8tion.jda.api.entities.ActionMetadata;
+import net.dv8tion.jda.api.entities.TextChannel;
+
+import java.util.Objects;
+
+public class ActionMetadataImpl implements ActionMetadata {
+
+    private TextChannel channel;
+    private int duration;
+
+    @Override
+    public TextChannel getChannel() {
+        return channel;
+    }
+
+    @Override
+    public int getDuration() {
```

Comment:
Should either return a `java.time.Duration` object or renamed to `getDurationSeconds`. See prior comment on `ActionMetadata`